File: py/test_pareto.py
import numpy as np
import pareto_front as pf
import matplotlib.pyplot as plt

x = np.array([[1,2], [2, 3], [4, 1], [1, 1], [2,2], [3, 2.5]])

pfismin = pf.is_pareto_efficient_indexed(x, True)
print(x[pfismin])

pfismax = pf.is_pareto_efficient_indexed(x, False)
print(x[pfismax])

plt.figure()
plt.scatter(x[:,0], x[:,1])
plt.show()

File: py/test_pareto.jl
using MultiObjective
using PyPlot

x = Array([1 2; 2 3; 4 1; 1 1; 2 2; 3 2.5]')

pfismin = MultiObjective.is_pareto_efficient(x, true)
print(x[:,pfismin])

pfismax = MultiObjective.is_pareto_efficient(x, false)
print(x[:,pfismax])

figure()
scatter(x[1,:], x[2,:])
show()
